nemo/collections/tts/models/fastpitch.py

```diff
@@ -142,13 +144,15 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         speaker_emb_condition_prosody = cfg.get("speaker_emb_condition_prosody", False)
         speaker_emb_condition_decoder = cfg.get("speaker_emb_condition_decoder", False)
         speaker_emb_condition_aligner = cfg.get("speaker_emb_condition_aligner", False)
+        global_style_token_encoder = instantiate(self._cfg.get("global_style_token", None))
```

Comment:
Is there any chance that the changes inside of the `FastPitchModel` class with regard to all these GST components can be moved to the `FastPitchAdapterModelMixin` class instead? How will the `FastPitchModel` class work if we end up wanting to add SV embedding instead of GST?

Reply:
`FastPitchAdapterModelMixin` class has functionalities related to adapters only. GST component is for speaker representation, so moving this to `FastPitchAdapterModelMixin` class will not be the right thing to do. What I have in mind is to have a speaker representation component (just like other components like pitch predictor, duration predictor etc) which can be initialized from the config. That way we can use GST or SV embedding or lookup embeddings or anything else based on the config. This would mean the config would have a section/struct for speaker representation like `model.speaker_embedding`.

Reply:
I need to check compatibility of this approach with already published models.

Reply:
We need a way of separating adapted FastPitch model from non-adapted Fastpitch models. I think we should consider a new class for this PR.

Reply:
how is the result？ I do this exp，but the result is bad